src/execution/operator/helper/physical_limit.cpp

```diff
@@ -13,14 +13,32 @@ namespace duckdb {
 class LimitGlobalState : public GlobalSinkState {
 public:
 	explicit LimitGlobalState(const PhysicalLimit &op) : current_offset(0) {
-		this->limit = op.limit_expression ? INVALID_INDEX : op.limit_value;
-		this->offset = op.offset_expression ? INVALID_INDEX : op.offset_value;
+		if (!op.limit_expression) {
+			this->limit = op.limit_value;
```

Comment:
Please create a `PhysicalLimitPercent` that runs the limit with a percentage and revert any changes made to the physical limit. This switching on two entirely different modes of execution inside one operator is not ideal.

Reply:
Okay, I will add PhysicalLimitPercent.